modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2533,12 +2533,24 @@ def _callable_func(self, func, axis, *args, **kwargs):
     groupby_sum = GroupByReduce.register("sum")
 
     def groupby_mean(self, by, axis, groupby_kwargs, agg_args, agg_kwargs, drop=False):
+        by_columns = by
+        if isinstance(by, BaseQueryCompiler):
```

Comment:
is it possible for `by` to be a normal pandas objects?

Reply:
We normally don't allow that kind of mixing pandas and modin objects. However, it looks like this case defaults to pandas later in this function:

```python
import modin.pandas as pd

df = pd.DataFrame([[1, 2], [3, 4]])
print(df.groupby(0).mean())
print(df.groupby(df._to_pandas()[0]).mean())
```

... but it's also possibly buggy (results for pandas by vs modin by are different). Anyway I don't think we need to worry about it here.